server/src/core/payment/tests/validator.test.ts

```
@@ -0,0 +1,113 @@
+    testValidationSucceeds(validators.validatesGetAllByUser, [
+      generateId(),
+    ]);
+  });
+  it('should throw error for all invalid test cases', () => {
+    testValidationFails(validators.validatesGetAllByUser, [
+      123832929223627,
+      'dksllcjzcsdmqozm123E2OE',
+      'jkshvfb,cnsg&"àéà"ç2201393332EECDKRU842823.KSkflkfvvnfrozccDHDCJQvnfvvze32E9ZEZIZRZRZOR391029Egjkvkvdfc ',
+    ]);
+  });
+})
+
+describe('validatesInitiateDonation', () => {
+  it('should not throw error if all test cases are valid', () => {
+    testValidationSucceeds(validators.validatesInitiateDonation, [
+      { userId: generateId(), amount: 100 },
+      { userId: generateId(), amount: 500 },
+      { userId: generateId(), amount: 2500 },
+      { userId: generateId(), amount: 10905 },
+      { userId: generateId(), amount: 527350 },
+    ]);
+  });
+  it('should throw error for all invalid test cases', () => {
+    testValidationFails(validators.validatesInitiateDonation, [
+      { userId: 'fhdgjqjqmqcqmlcq', amount: 100 },
+      { userId: '', amount: 100 },
+      { userId: 34256, amount: 2000 },
+      { userId: {}, amount: 530 },
+      { amount: 100 },
+      { userId: generateId() },
+      { userId: generateId(), amount: 10 },
+      { userId: generateId(), amount: '1000' }
+    ]);
+  });
+})
+
+describe('validatesSendDonation', () => {
+  it('should not throw error if all test cases are valid', () => {
+    testValidationSucceeds(validators.validatesSendDonation, [
+      { from: generateId(), to: generateId(), amount: 100 },
+      { from: generateId(), to: generateId(), amount: 20 },
+      { from: generateId(), to: generateId(), amount: 0 },
+      { from: generateId(), to: generateId(), amount: 500 },
+      { from: generateId(), to: generateId(), amount: 2000 },
+    ]);
+  });
+  it('should throw error for all invalid test cases', () => {
+    testValidationFails(validators.validatesSendDonation, [
```

Should throw an error if the argument includes unexpected fields